test/e2e_test.go

```diff
@@ -165,3 +221,55 @@ func TestE2EPutGetLogicForMemoryStore(t *testing.T) {
 	require.NoError(t, err)
 	require.Equal(t, testPreimage, preimage)
 }
+
+func TestMemStoreWithProxyClient(t *testing.T) {
```

Comment:
nit: Theoretically it should be possible to dry up these tests. Not necessary to do manually but maybe chatgpt can do.